lambda_function.py

```diff
@@ -0,0 +1,629 @@
+    validate_event_domainname(data)
+    validate_event_email(data)
+
+def validate_event_action(data):
+    '''Check if request has valid Action key.'''
+    # Action unknown
+    if data.get('Action') not in ['CREATE', 'DELETE', 'UPDATE']:
+        #log.error('Action is not valid')
+        print(json.dumps({'message': 'Action is not valid'}, default=str))
+        sys.exit('Action is not valid')
+
+def validate_event_keys(data):
+    '''Check if request has mandatory keys.'''
+    # Not mandatory 'NameServers' handler for 'Action': 'DELETE'
+    if data['Action'] == 'DELETE' and 'NameServers' not in data:
+        data['NameServers'] = 'fake1,fake2'
+    for key in ['Action', 'UserName', 'NameServers', 'ZoneID', 'DomainName', 'Email']:
+        if key not in data:
+           #log.error('Key: "{}" is not set'.format(key))
+            print(json.dumps({'message': 'Key is not set', 'key': key}, default=str))
+            sys.exit('Key: "{}" is not set'.format(key))
+
+def validate_event_username(data):
+    '''Check if request has valid UserName key.'''
+    key = 'UserName'
+    regexp = re.match(r'^[0-9a-zA-Z\-_]+$', data[key])
+    if not regexp:
+        sys.exit('{} value is not valid: "{}"'.format(key, data[key]))
+
+def validate_event_nameservers(data):
+    '''Check if request has valid NameServers key.'''
+    key = 'NameServers'
+    regexp = re.match(r'^[0-9a-zA-Z\-\.,]+$', data[key])
+    if not regexp:
+        sys.exit('{} value is not valid: "{}"'.format(key, data[key]))
+
+def validate_event_zoneid(data):
+    '''Check if request has valid ZoneID key.'''
+    key = 'ZoneID'
+    regexp = re.match(r'^[0-9a-zA-Z]+$', data[key])
+    if not regexp:
+        sys.exit('{} value is not valid: "{}"'.format(key, data[key]))
+
+def validate_event_domainname(data):
+    '''Check if request has valid DomainName key.'''
+    key = 'DomainName'
+    regexp = re.match(r'^[0-9a-zA-Z\-\.]+$', data[key])
+    if not regexp:
+        sys.exit('{} value is not valid: "{}"'.format(key, data[key]))
+
+def validate_event_email(data):
+    '''Check if request has valid Email key.'''
+    key = 'Email'
+    regexp = re.match(r'^[0-9a-zA-Z\-\.@]+$', data[key])
+    if not regexp:
+        sys.exit('{} value is not valid: "{}"'.format(key, data[key]))
+
+def data_from_event_to_config(event):
+    '''Add data from "event" to dictionary with config options.'''
+    for key in event['body']:
+        value = event['body'][key]
+        conf[key] = value
+    if not conf['DomainName'].endswith('.'):
+        conf['DomainName'] += '.'
+    conf['NameServers_list'] = conf['NameServers'].split(',')
+    for index in range(len(conf['NameServers_list'])):
+        ns = conf['NameServers_list'][index]
+        if not ns.endswith('.'):
+            conf['NameServers_list'][index] += '.'
+    conf['source_ip'] = event['headers']['X-Forwarded-For']
+    #log.debug('Config content: {}'.format(conf))
+    print(json.dumps(conf, default=str))
```

Comment:
Please, add 'air' before and after constructions like def, if, for time to time - it make code more readable

```suggestion
    for key in event['body']:
        value = event['body'][key]
        conf[key] = value
    
    if not conf['DomainName'].endswith('.'):
        conf['DomainName'] += '.'
    
    conf['NameServers_list'] = conf['NameServers'].split(',')
    
    for index in range(len(conf['NameServers_list'])):
        ns = conf['NameServers_list'][index]
        if not ns.endswith('.'):
            conf['NameServers_list'][index] += '.'
    
    conf['source_ip'] = event['headers']['X-Forwarded-For']
    #log.debug('Config content: {}'.format(conf))
    print(json.dumps(conf, default=str))
```

Reply:
Stile changes - please write policy for it.

Reply:
PEP-8 mandatory, pylint - recomended

https://www.python.org/dev/peps/pep-0008/
https://docs.pylint.org/en/1.6.0/tutorial.html